### PR TITLE
fix(HBO Max): fix episode/movie name

### LINKED ISSUE #6495: HBO Max | https://play.hbomax.com
### Steps to Reproduce

1. Watch a normal series/episode

2. Status shows "Watching movie" instead of the episode name

### Expected Behavior

Status is supposed to show the episode name & season, show and how many minutes are left

### Actual Behavior

It would show "Watching Movie" when I'm not watching a movie.

### Page URL

https://play.hbomax.com/

### Screenshots

Expected
image

Actual
image

### Operating System

Windows 11

### Browser and Version

Brave Browser V1.41.100 

### Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Resolves #6495
Fixed episode/movie name by changing the element for temp use while waiting for the fetch.

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/H/HBO Max/presence.ts [hunk 1]
```diff
@@ -126,14 +127,39 @@ async function fetchCover() {
...
+			}
+		).then(res => res.json());
+
+		titles = response[0].body.titles.short;
+	} catch {
+		return;
+	}
+
+	isFetching = false;
+}
```
Comment (Slowlife01) [inline websites/H/HBO Max/presence.ts:154]: Can't you just edit `fetchCover()`? And you do realize `isFetching` wouldn't do anything here right

File: websites/H/HBO Max/presence.ts [hunk 2]
```diff
@@ -149,18 +175,17 @@ presence.on("UpdateData", async () => {
...
-				document.querySelectorAll("[role=heading]:first-child span span")
-			)
-				.map(z => z.textContent)
-				.filter(z => z.length > 1 && !/\d \/ \d+/.test(z))), // Test for "d / d" ex.: 01:45 / 01:30:00
-				(hasEpisode = titles.length > 1);
-
+			titles = document
+				.querySelector('[href*="type:episode"]')
+				?.getAttribute("aria-label");
+			await fetchTitle();
```
Comment (Slowlife01) [inline websites/H/HBO Max/presence.ts:181]: You should only call this once..

File: websites/H/HBO Max/presence.ts [hunk 3]
```diff
@@ -190,7 +215,8 @@ presence.on("UpdateData", async () => {
 					startTimestamp: timestamps[0],
 					endTimestamp: timestamps[1],
 				});
-			}
+			} else delete presenceData.endTimestamp;
```
Comment (Slowlife01) [inline websites/H/HBO Max/presence.ts:218]: I'm pretty sure this is useless since endTimestamp isn't set anywhere before this

**Review by Slowlife01 — COMMENTED** [review]
Also please move episode number and stuff to the bottom instead, just where it was before. And you should be able to get the episode name with the api?

File: websites/H/HBO Max/presence.ts (hunk 1 above)
Comment (DarkVillager) [reply websites/H/HBO Max/presence.ts:154]: Ok

**Comment by EncryptedDev** [comment]
@Slowlife01, you happy with these changes?😄

File: websites/H/HBO Max/presence.ts [hunk 4]
```diff
@@ -179,7 +170,10 @@ presence.on("UpdateData", async () => {
 					const episodeId = location.pathname.match(/:episode:([^:]+)/)[1];
 
 					if (isFetching) return;
-					coverUrls[episodeId] ??= await fetchCover();
+					await fetchCoverAndTitle();
```
Comment (Slowlife01) [inline websites/H/HBO Max/presence.ts:173]: Well, you are still spamming the request

Comment (DarkVillager) [reply websites/H/HBO Max/presence.ts:173]: > Well, you are still spamming the request

it only runs once i think lemme check.

**Comment by DarkVillager** [comment]
Thanks @EncryptedDev for removing the console log 😂

**Comment by DarkVillager** [comment]
@Slowlife01 how about now?

**Comment by Slowlife01** [comment]
Well, this is quite messed up actually. May I create a new PR, if you don't mind?

**Comment by DarkVillager** [comment]
> Well, this is quite messed up actually. May I create a new PR, if you don't mind?

sure

**Pull request closed without merging**